### PR TITLE
test: fix daser test

### LINKED ISSUE #921: test: flaky TestDASer_stopsAfter_BEFP
Failed on main
```
=== RUN   TestDASer_stopsAfter_BEFP
2022-07-18T20:20:29.366Z	ERROR	share	share/full_availability.go:62	availability validation failed	{"root": "45zl/3Xr/Ssybz1uKOqgXrDww5ox/GSyUK4F5h9l7aQ=", "err": "byzantine error. Axis:1, Index:0"}
2022-07-18T20:20:29.366Z	ERROR	das	das/daser.go:275	sampling failed	{"height": 1, "hash": "60AF2EB2227EF10770CF65484544FF9E64F43FB5188B699AA18359CC8642E129", "square width": 4, "data root": "45zl/3Xr/Ssybz1uKOqgXrDww5ox/GSyUK4F5h9l7aQ=", "err": "byzantine error. Axis:1, Index:0"}
2022-07-18T20:20:29.366Z	ERROR	das	das/daser.go:210	catch-up routine failed	{"attempted range: from": 0, "to": 15}
2022-07-18T20:23:24.253Z	ERROR	fraud	fraud/subscription.go:52	reading next proof failed	{"err": "context deadline exceeded"}
    daser_test.go:290: context deadline exceeded
--- FAIL: TestDASer_stopsAfter_BEFP (180.00s)
```

### PR DESCRIPTION
**Pull request by vgonkivs**
Resolves #921

The main reason in flakiness is that BEFP was broadcasted before daser was subscribed to it.

### CODE REVIEW

**Review by Wondertan — APPROVED** [review]
Try running the test with -count 100, and runtime.GOMAXPROCS(1) to ensure it never happens locally

**Review by Wondertan — APPROVED** [review]
utACK

File: das/daser_test.go [hunk 1]
```diff
@@ -274,20 +274,23 @@ func TestDASer_stopsAfter_BEFP(t *testing.T) {
...
 	mockGet.headers[1] = header.CreateFraudExtHeader(t, mockGet.headers[1], bServ)
-
+	newCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
 	// create and start DASer
 	daser := NewDASer(shareServ, sub, mockGet, ds, f)
-
-	require.NoError(t, daser.Start(ctx))
-	fraud.OnProof(ctx, f, fraud.BadEncoding,
+	ch := make(chan struct{})
```
Comment (renaynay) [inline das/daser_test.go:281]: `fraudCh` pls

File: das/daser_test.go [hunk 2]
```diff
@@ -274,20 +274,23 @@ func TestDASer_stopsAfter_BEFP(t *testing.T) {
...
-	// wait for dasing catch-up routine fails
+	time.Sleep(time.Millisecond * 100)
+	require.NoError(t, daser.Start(newCtx))
+	// wait for fraud proof will be handled
 	select {
 	case <-ctx.Done():
 		t.Fatal(ctx.Err())
-	case <-daser.ctx.Done():
+	case <-ch:
+		require.NoError(t, daser.Stop(ctx))
```
Comment (renaynay) [inline das/daser_test.go:293]: Doesn't this not actually test that the daser is stopped by the BEFP handling?

Comment (vgonkivs) [reply das/daser_test.go]: reworked

**Comment by vgonkivs** [comment]
one more flakiness test 😢 

```
--- PASS: TestFullReconstructFromBridge (1.16s)
=== RUN   TestFullReconstructFromLights
panic: Fail in goroutine after TestFullReconstructFromBridge has completed
goroutine 19355 [running]:
testing.(*common).Fail(0xc003b189c0)
       /opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:817 +0xe5
testing.(*common).Errorf(0xc003b189c0, {0x22898b1?, 0xc003b189c0?}, {0xc0089bb800?, 0x47822f?, 0xc002c28f90?})
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:934 +0x65
github.com/stretchr/testify/assert.Fail({0x2f1f3a0, 0xc003b189c0}, {0xc0040c4120, 0x84}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:264 +0x384
github.com/stretchr/testify/assert.NoError({0x2f1f3a0, 0xc003b189c0}, {0x2f1f0c0, 0xc004f7b290}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:1392 +0x105
github.com/stretchr/testify/require.NoError({0x2f27150, 0xc003b189c0}, {0x2f1f0c0, 0xc004f7b290}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.0/require/require.go:1261 +0x96
github.com/celestiaorg/celestia-node/node/tests/swamp.(*Swamp).SubmitData(0xc003d57830, {0x2f3b2a0, 0xc004baaa80}, 0x0?, {0xc004393000, 0x1000, 0x100c000ed7d38?})
	/home/runner/work/celestia-node/celestia-node/node/tests/swamp/swamp_tx.go:19 +0x105
github.com/celestiaorg/celestia-node/node/tests/swamp.(*Swamp).FillBlocks(0xc003d57830, {0x2f3b2a0, 0xc004baaa80}, 0xc004022e40?, 0x10, 0x14)
	/home/runner/work/celestia-node/celestia-node/node/tests/swamp/swamp_tx.go:31 +0x177
created by github.com/celestiaorg/celestia-node/node/tests.TestFullReconstructFromBridge
	/home/runner/work/celestia-node/celestia-node/node/tests/reconstruct_test.go:47 +0x14b
FAIL	github.com/celestiaorg/celestia-node/node/tests	17.078s
FAIL
```

**Review by renaynay — APPROVED** [review]
morge

**Pull request merged**